project/ValidatePullRequest.scala

```diff
@@ -15,6 +15,38 @@ import scala.util.matching.Regex
 
 object ValidatePullRequest extends AutoPlugin {
 
+  sealed trait BuildMode {
+    val Zero = Def.task { () } // when you stare into the void, the void stares back at you
+    def task: Def.Initialize[Task[Unit]]
+    def log(projectName: String, l: Logger): Unit
+  }
+  case object BuildSkip extends BuildMode {
+    override def task = Zero
+    def log(projectName: String, l: Logger) =
+      l.info(s"Skipping validation of [$projectName], as PR does NOT affect this project...")
+  }
+  case object BuildQuick extends BuildMode {
+    override def task = Zero.dependsOn(test in ValidatePR)
+    def log(projectName: String, l: Logger) = 
+      l.info(s"Building [$projectName] in quick mode, as it's dependencies were affected by PR.")
+  }
+  case object BuildProjectChangedQuick extends BuildMode {
+    override def task = Zero.dependsOn(test in ValidatePR)
+    def log(projectName: String, l: Logger) =
+      l.info(s"Building [$projectName] as the root `project/` directory was affected by this PR.")
+  }
+  case object BuildForcedAll extends BuildMode {
```

Comment:
Good catch, thanks - the one used is `BuildCommentForcedAll` - has a bit more info. I'll remove and merge